src/utilities/schemaComparisons.js

```diff
@@ -0,0 +1,220 @@
+/* @flow */
+/**
+ *  Copyright (c) 2016, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import {
+  getNamedType,
+  GraphQLEnumType,
+  GraphQLInputObjectType,
+  GraphQLInterfaceType,
+  GraphQLObjectType,
+  GraphQLUnionType,
+} from '../type/definition';
+import type {
+  GraphQLFieldDefinitionMap,
+  GraphQLNamedType,
+  GraphQLType,
+  InputObjectFieldMap,
+} from '../type/definition';
+import {
+  GraphQLSchema,
+} from '../type/schema';
+import type {
+  TypeMap,
+} from '../type/schema';
+
+/**
+ * Given two schemas, returns a Set containing descriptions of all the types of
+ * breaking changes covered by the other functions down below.
+ */
+export function findBreakingChanges(
+  oldSchema: GraphQLSchema,
+  newSchema: GraphQLSchema
+): Set<string> {
+  return new Set([
```

Comment:
What's the reasoning behind using Sets throughout instead of just Arrays?

It seems like there ends up being a lot of work converting between Sets and Arrays and Object maps and we should ensure it's worth doing all that.

Also, I bet users of this function will expect an array of errors rather than a Set.

It's cool to use Set as an implementation detail in deriving these breakages though!

e.g. this could become:

```js
return [
  ...findRemovedTypes(oldSchema, newSchema),
  ...findTypesThatChangedType(oldSchema, newSchema),
  ...findBreakingFieldChanges(oldSchema, newSchema),
  ...findTypesRemovedFromUnions(oldSchema, newSchema),
  ...findValuesRemovedFromEnums(oldSchema, newSchema)
];
```